### PR TITLE
Add check for default docker desktop endpoint on linux

### PR DESCRIPTION
**Pull request by FPVian**
According to [this documentation](https://docs.docker.com/desktop/faqs/linuxfaqs/#context), the default endpoint for docker desktop on linux is `unix:///$HOME/.docker/desktop/docker.sock`. Right now, we are only checking the default location on MacOS, `unix:///$HOME/.docker/run/docker.sock`. This PR adds a check for this if the other checks for a default system-wide host and MacOS user space host fail.

Fixes [#720](https://github.com/pulumi/pulumi-docker/issues/720)

### CODE REVIEW

**Comment by thomas11** [comment]
Hey @FPVian, sorry for the delay. The PR looks good to me! Now that we're up to three endpoints to try, maybe it would be nicer to iterate over them instead of repeating the code three times. I'll leave that up to you.

**Review by thomas11 — APPROVED** [review]

**Comment by FPVian** [comment]
Okay I finally updated it to use a for loop. @thomas11 is this what you had in mind?

**Comment by mikhailshilkov** [comment]
@FPVian @thomas11 Can we move this PR to completion?

**Comment by FPVian** [comment]
@thomas11 Not at all, thanks for the help! I found an error in my original code where the function was returning early without an error. I changed the condition and can confirm this is working on my computer now.

**Comment by thomas11** [comment]
Thanks @FPVian! I'm good to merge this if the latest test run passes. I'd like to give @guineveresaenger a chance for a final look, too, though.

File: provider/image.go
Comment (guineveresaenger) [inline provider/image.go:817]: This is a future enhancement, because I'd like to get this pull request in: 
I would like it if we could surface the `testConnection` error directly here - not all logs always get surfaced in the most user-friendly way.

**Review by guineveresaenger — APPROVED** [review]
Apologies for the lateness - I'm at a conference this week.

This LGTM. Waiting on the final test run and then I'll merge.

Thank you so much for your contribution, @FPVian!

**Pull request merged**
